### PR TITLE
Nextflow integration with pipeline

### PR DESCRIPTION
**Pull request by ankurjuneja**
#### Rationale
This PR adds a new Nextflow module to easily initiate DIA (data independent analysis) through the LK pipelines, both pulling data from PanoramaWeb and uploading the results back into PanoramaWeb. 

#### Related Pull Requests
* 

#### Changes
* new Nextflow module
* Nextflow pipeline job

### CODE REVIEW

File: nextflow/src/org/labkey/nextflow/NextFlowController.java [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+            if (Boolean.parseBoolean(map.get(IS_NEXTFLOW_ENABLED)))
+            {
+                btnTxt = "Disable NextFlow";
+            }
+            else
+            {
+                btnTxt = "Enable NextFlow";
+            }
+
+            return new HtmlView("Enable/Disable Nextflow", DIV("Enable/Disable Nextflow",
```
Comment (labkey-jeckels) [inline nextflow/src/org/labkey/nextflow/NextFlowController.java]: These kinds of UIs can be confusing about what's the current vs future state.

In addition to the button, let's clearly state "NextFlow is currently enabled in this folder" (or disabled).

File: nextflow/src/org/labkey/nextflow/NextFlowController.java [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    @RequiresPermission(AdminOperationsPermission.class)
+    public class NextFlowRunAction extends FormViewAction
+    {
+        private ActionURL _successURL;
+        @Override
+        public void validateCommand(Object o, Errors errors)
+        {
+            PropertyStore store = PropertyManager.getNormalStore();
+            PropertyManager.PropertyMap map = store.getWritableProperties(NextFlowManager.NEXTFLOW_ENABLE, false);
+            if (map == null || !"true".equals(map.get("enabled")))
```
Comment (labkey-jeckels) [inline nextflow/src/org/labkey/nextflow/NextFlowController.java]: Boolean.parseBoolean()

File: nextflow/src/org/labkey/nextflow/NextFlowController.java [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+        @Override
+        public URLHelper getSuccessURL(Object o)
+        {
+            return PageFlowUtil.urlProvider(PipelineStatusUrls.class).urlBegin(getContainer());
+        }
+
+        @Override
+        public void addNavTrail(NavTree navTree)
+        {
+
```
Comment (labkey-jeckels) [inline nextflow/src/org/labkey/nextflow/NextFlowController.java]: Append to the nav trail to give the page a title

File: nextflow/src/org/labkey/nextflow/pipeline/NextFlowPipelineJob.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    protected NextFlowPipelineJob()
+    {}
+
+    public NextFlowPipelineJob(ViewBackgroundInfo info, @NotNull PipeRoot root, String apiKey)
+    {
+        super(null, info, root);
+        this._apiKey = apiKey;
+        try
+        {
+            setLogFile(FileUtil.createTempFile("NextFlowPipelineJob", ".log"));
```
Comment (labkey-jeckels) [inline nextflow/src/org/labkey/nextflow/pipeline/NextFlowPipelineJob.java]: This shouldn't be a temp file. Create a file based on the current date/time in the pipeline root. See `MaintenancePipelineJob` for an example.

**Review by labkey-jeckels — APPROVED** [review]
Some requests but no need for another review unless you want it.

File: nextflow/src/org/labkey/nextflow/NextFlowController.java (hunk 1 above)
Comment (ankurjuneja) [reply nextflow/src/org/labkey/nextflow/NextFlowController.java]: done

File: nextflow/src/org/labkey/nextflow/NextFlowController.java (hunk 2 above)
Comment (ankurjuneja) [reply nextflow/src/org/labkey/nextflow/NextFlowController.java]: done

File: nextflow/src/org/labkey/nextflow/NextFlowController.java (hunk 3 above)
Comment (ankurjuneja) [reply nextflow/src/org/labkey/nextflow/NextFlowController.java]: done

File: nextflow/src/org/labkey/nextflow/pipeline/NextFlowPipelineJob.java (hunk 4 above)
Comment (ankurjuneja) [reply nextflow/src/org/labkey/nextflow/pipeline/NextFlowPipelineJob.java]: done

**Pull request merged**